1.6/windows/nanoserver/Dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM microsoft/nanoserver
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop';"]
+
+# ideally, this would be C:\go to match Linux a bit closer, but C:\go is the recommended install path for Go itself on Windows
+ENV GOPATH C:\\gopath
+
+# PATH isn't actually set in the Docker image, so we have to set it from within the container
+RUN Write-Host 'Updating PATH ...'; \
+	$regPath = 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment'; \
+	\
+	$oldPath = ( \
+	Get-ItemProperty \
+		-Path $regPath \
+		-Name PATH \
+	).path; \
+	Write-Host ('  Before: {0}' -f $oldPath); \
+	\
+	$newPath = $env:GOPATH + '\bin;C:\go\bin;' + $oldPath; \
+	Write-Host ('  After:  {0}' -f $newPath); \
+	\
+	Set-ItemProperty \
+		-Path $regPath \
+		-Name PATH \
+		-Value $newPath; \
+	Write-Host 'Complete.';
+# doing this first to share cache across versions more aggressively
+
+ENV GOLANG_VERSION 1.6.3
+ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.windows-amd64.zip
+ENV GOLANG_DOWNLOAD_SHA256 6a18e5ed8b39785338986aecc6a3f36f5c4be286ff52db0ae3bcd2275ab70df0
+
+RUN Write-Host ('Downloading {0} ...' -f $env:GOLANG_DOWNLOAD_URL); \
+	Invoke-WebRequest -Uri $env:GOLANG_DOWNLOAD_URL -OutFile 'go.zip'; \
```

Comment:
@tianon I found your measurements again. Today I also tried to build go1.8beta1 for Windows, but had timeouts using `Invoke-WebRequest`, so I switched to `DownloadFile()`. I also tested it in my Win2016 host and download with `Invoke-WebRequest` took over 5 minutes and with `DownloadFile()` only about 10 seconds.

I then put Fiddler in between. The GET request in Fiddler was finished after about 9.9 seconds,  and THEN PowerShell `Invoke-WebRequest` showed me the slow progress bar and high CPU load. Using `DownloadFile()` it went through Fiddler in 10 seconds.

So I come to the conclusion that `Invoke-WebRequest` still should be avoided at least for windowsservercore images.

<img width="1003" alt="golang-invoke-webrequest-skitch" src="https://cloud.githubusercontent.com/assets/207759/20805843/b6c21fce-b7f8-11e6-935b-156f5ea0d288.png">

/cc @friism